resilience4j-prometheus/src/main/java/io/github/resilience4j/prometheus/Histograms.java

```diff
@@ -0,0 +1,273 @@
+package io.github.resilience4j.prometheus;
+
+import io.prometheus.client.Histogram;
+import javaslang.control.Try;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static io.github.resilience4j.prometheus.HistogramsCompanion.startFailureTimer;
+import static io.github.resilience4j.prometheus.HistogramsCompanion.startSuccessTimer;
+
+/**
+ * The set of function decorators to measure time of function execution and fill Prometheus {@link Histogram}
+ *
+ * Supplied {@link Histogram} instance should contain 1 label. The label will have 2 values: "success" and "failure"
+ * distinguishing successful and failed executions correspondingly.
+ */
+public interface Histograms {
+    /**
+     * Decorates and executes the decorated Runnable.
+     *
+     * @param histogram the histogram to use
+     * @param runnable the original Callable
+     */
+    static void executeRunnable(Histogram histogram, Runnable runnable) throws Exception {
```

Comment:
When we have a TimerContext class, you can change this from `static` to `default`  and remove the Histogram parameter.